### DISCUSSION TITLE
[SOLVED] How to enforce top aligned terminal prompts in horizontal and vertical splits

### DISCUSSION THREAD

**Discussion by t-391** [i3/i3#5394] [Q&A]
Edit: I spoke with a developer on the rxvt-unicode team. He confirmed that the behavior I observed below is not related to i3-wm, it is a function of rxvt-unicode's handling of scrollback buffers & window resizing responses that caused it. For now, anyone experiencing this issue can roll back to V9.30 to retain the old behavior. For the future, I am told the team is considering adding a flag that would allow for both behaviors to be options.

Hello everyone,

I recently updated to i3-wm 4.22. It caused an unexpected behavior. 

Previously, **regardless of how I split my windows, horizontal or vertical**, the PS1 ZSH Prompt always aligned to the top of the window.

Now,  when I split my windows horizontally, the PS1 ZSH Prompts center inside the windows (unwanted behavior): [Imgur](https://i.imgur.com/Nr4QpfF.png)

When I split my windows vertically, the PS1 ZSH Prompts still align to the top of the window (desired behavior): [Imgur](https://i.imgur.com/Hda2Au5.png)

Before updating, my terminal prompts always aligned to the top of the window, whether vertical or horizontal split.
How can I set my i3 config so prompts always align to the top of the window?

My .zshrc files have not changed in a very long time, so I do not suspect another program is causing the behavior.

Edit: Perhaps I should say, when I issue a `clear` command to the terminal, the prompt resets to the top of the screen, as desired. But when spawned, in horizontal split mode, it is centered.

**Comment by orestisfl** [comment]
Can you please downgrade i3 and check that the desired behavior still happens with the old version?

**Reply by t-391** [reply]
Hello, I tried both 4-21 and 4-20, and the bad behavior persists in both versions. I think this means the issue must exist outside of i3-wm, would you agree?

**Reply by okraits** [reply]
Which terminal do you use? Does it happen with another terminal as well?

**Reply by t-391** [reply]
Thanks for this idea. I use Urxvt, so as a test I installed terminator, and everything works. So I'll check the changelogs of rxvt-unicode to see if a new default setting has changed. I will update my main post with the solution if I can find it.
